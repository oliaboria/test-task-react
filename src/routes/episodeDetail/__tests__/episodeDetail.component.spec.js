import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import ErrorMessage from '../../../components/errorMessage';
import Loader from '../../../components/loader';
import mockEpisodeDetail from '../../../mocks/episodeDetail';
import EpisodeDetail from '../episodeDetail.component';
import useAsyncEpisodeDetail from '../episodeDetail.hooks';

jest.mock('../episodeDetail.hooks');

describe('Episode Detail page', () => {
    it('should render loading state', () => {
        useAsyncEpisodeDetail.mockImplementation(() => ({
            episode: null,
            isEpisodeLoading: true,
            isEpisodeLoadingFail: false,
        }));

        const wrapper = shallow(<EpisodeDetail />);
        expect(wrapper.find(Loader)).toHaveLength(1);
    });

    it('should render error state', async () => {
        useAsyncEpisodeDetail.mockImplementation(() => ({
            episode: null,
            isEpisodeLoading: false,
            isEpisodeLoadingFail: true,
        }));

        const wrapper = shallow(<EpisodeDetail />);
        expect(wrapper.find(ErrorMessage)).toHaveLength(1);
    });

    it('should render card component', () => {
        useAsyncEpisodeDetail.mockImplementation(() => ({
            episode: mockEpisodeDetail,
            isEpisodeLoading: false,
            isEpisodeLoadingFail: false,
        }));

        const wrapper = shallow(<EpisodeDetail />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
