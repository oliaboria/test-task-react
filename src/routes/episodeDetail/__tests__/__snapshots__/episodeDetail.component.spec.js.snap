// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode Detail page should render card component 1`] = `
<Fragment>
  <div
    className="episode-detail"
  >
    <Card
      card={
        Object {
          "_links": Object {
            "self": Object {
              "href": "http://api.tvmaze.com/episodes/657309",
            },
          },
          "airdate": "2016-04-04",
          "airstamp": "2016-04-04T22:00:00+00:00",
          "airtime": "18:00",
          "id": 657309,
          "image": Object {
            "medium": "http://static.tvmaze.com/uploads/images/medium_landscape/53/132618.jpg",
            "original": "http://static.tvmaze.com/uploads/images/original_untouched/53/132618.jpg",
          },
          "name": "Princess Buttercup",
          "number": 2,
          "runtime": 15,
          "season": 1,
          "summary": "<p>Buttercup becomes enamored with a team of roller derby girls called the Derbytantes. When she starts spending all of her time with them, Princess Morbucks seeks to be her replacement in the Powerpuff Girls.</p>",
          "url": "http://www.tvmaze.com/episodes/657309/the-powerpuff-girls-1x02-princess-buttercup",
        }
      }
    />
  </div>
</Fragment>
`;
